components/base/sensorcontrolled/movestraight.go

```diff
@@ -77,6 +90,10 @@ func (sb *sensorBase) MoveStraight(
 		}
 	}
 
+	// this state is only used when no position sensor is configured
+	prevTime := startTime
+	currDistMm := 0.
```

Comment:
not sure if theres a better spot to put this state that only functions during the `position == nil` case, or if I should try to refactor to make the split more obvious

Reply:
I think it's fine. the separation of behaviors is a little foggy but the comments are enough in my opinion